content/reference/wadm/api.md

```diff
@@ -0,0 +1,252 @@
+specify the version of the model to be displayed. A list of stored versions can be obtained with the `versions` verb.
+
+**Request**:
+```json
+{
+    "version": "1.0"
+}
+```
+
+**Response**:
+
+JSON serialization of [OAM model/yaml](https://github.com/wasmCloud/wadm/tree/main/oam)
+
+
+
+### Store Models
+`wadm.api.{lattice}.model.put.{name}`
+
+Stores the named model. If the model and version being submitted already exist, the request will be rejected. New versions will be appended to the history according to retention policy. Note that this won't automatically deploy a model, it only affects storage. The response will tell the caller how many versions are on file and the current version number _after_ the operation completed.
+
+**Request**: JSON serialization of OAM model
+
+**Response**:
+```json
+{
+    "result": "error|created|newversion",
+    "total_versions": 4,
+    "current_version": "1.0",
+    "message": "..."
+}
+```
+
+The response will indicate if storage failed, if a new model was created, or if a new version of a model was pushed.
+
+
+### Delete Models
+`wadm.api.{lattice}.model.del.{name}`
+
+Deletes the named model with the supplied version number. If the `delete_all` flag is set, then all versions of this model will be deleted. If the `undeploy` flag is true, then a corresponding `undeploy` operation will be performed on this model.
+
+The response indicates success or failure, and if an `undeploy` operation was requested as part of the delete.
+
+**Request**: 
+```json
+{
+    "version": "1.0",
+    "delete_all": false,
+    "undeploy": false
+}
+```
+
+**Response**:
+```json
+{
+    "result": "deleted|error",
+    "undeploy": true,
+    "message": "..."
+}
+```
+
+### Version History
+`wadm.api.{lattice}.model.versions.{name}`
+
+Retrieves a list of all stored versions of a given model, including when that version was created. 
+
+**Request**: empty
+
+**Response**: 
+```json
+[
+    {
+        "version": "...",
+        "created": "<RFC3339 Timestamp>",
+        "deployed": false
+    }
+]
+```
+
+Each of the items in the response list describes a revision and indicates whether that revision is the deployed one.
+
+## Managing Deployments
+Deployments are discrete instances of autonomous control agents that monitor an application model against the real-time, discovered state of a lattice. This agent monitors lattice events, compares lattice state against the desired state of the model, and issues lattice control interface commands in order to compensate for observed gaps.
+
+### Deploy
+`wadm.api.{lattice}.model.deploy.{name}`
+
+Deploys the indicated model. This operation is idempotent, and will simply "succeed" if a request to deploy an already deployed model is received. The previous version, if deployed, does not need to be _undeployed_. Instead, the desired state of the new version is compared against the observed state of the lattice and the autonomous agents will compensate accordingly.
+
+**Request**:
+```json
```

Comment:
ergonomic suggestion: allow "latest" for version name - (or interpret missing version to mean later) since this is probably the most common use case.